### PR TITLE
Fix flaky DBLock tests

### PR DESCRIPTION
**Pull request by nmarton-da**
As it turned out normal connection closeing does not imply released resources on the database side

* Adding synchronous eventually support
* use that for testing connection lost cases
* refactor a case to use proper release instead of connection close
* Increases a timeout to lower risk of flakyness

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsDBLock.scala [hunk 1]
```diff
@@ -153,4 +154,20 @@ private[backend] trait StorageBackendTestsDBLock extends Matchers with StorageBa
       Future.successful(1 shouldBe 1)
     }
   }
+
+  def eventually[T](t: => T): T = {
```
Comment (nmarton-da) [inline ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsDBLock.scala:158]: seems like something we might have somewhere. If we do please come forward

File: ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsDBLock.scala [hunk 2]
```diff
@@ -86,29 +87,29 @@ private[backend] trait StorageBackendTestsDBLock extends Matchers with StorageBa
...
+      backend.release(shared3)(c(3))
       backend.tryAcquire(backend.lock(1), LockMode.Exclusive)(c(4)) should not be empty
   }
 
   it should "lock immediately, or fail immediately" in dbLockTestCase(2) { c =>
     backend.tryAcquire(backend.lock(1), LockMode.Shared)(c(1)) should not be empty
     val start = System.currentTimeMillis()
     backend.tryAcquire(backend.lock(1), LockMode.Exclusive)(c(2)) shouldBe empty
-    (System.currentTimeMillis() - start) should be < 100L
+    (System.currentTimeMillis() - start) should be < 500L
```
Comment (nmarton-da) [inline ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsDBLock.scala:112]: could be another source of flakyness

File: ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsDBLock.scala (hunk 1 above)
Comment (cocreature) [inline ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsDBLock.scala:158]: I’m a bit confused why you need this. Why does scalatest’s `eventually` not do the trick? That should work for both futures and non-future stuff.

**Review by cocreature — APPROVED** [review]
Thanks! I don’t quite understand why you need your own `eventually` but no big issue defining it.

File: ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsDBLock.scala (hunk 1 above)
Comment (nmarton-da) [reply ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsDBLock.scala]: no confusion, I thought that is only for futures, thanks! changed it!

**Review by mziolekda — APPROVED** [review]

**Pull request merged**